src/game/attack/AttackExecutor.ts

```diff
@@ -20,11 +21,13 @@ export class AttackExecutor {
 	 * @param player The player that is attacking.
 	 * @param target The player that is being attacked, or null if the target is unclaimed territory.
 	 * @param troops The amount of troops that are attacking.
+	 * @param borderTiles The tiles from which the attack is executed. If it is null, it will be the player's border tiles by default.
 	 */
-	constructor(player: Player, target: Player | null, troops: number) {
+	constructor(player: Player, target: Player | null, troops: number, borderTiles: Set<number> | null) {
 		this.player = player;
 		this.target = target;
 		this.troops = troops;
+		this.borderTiles = borderTiles;
```

Comment:
This can be passed directly to orderTiles here (without adding it as a property to this class)